### PR TITLE
Update docs to use named links and remove numbers from paths

### PR DESCRIPTION
**Pull request by hasheddan**
### Description of your changes

Docs updates that fix breaking relative links and make the url paths not as gross.

### How has this code been tested?

`make reviewable`

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [x] Updated any relevant [documentation] and [examples].
- [ ] Reported all new error conditions into the log or as an event, as
  appropriate.

For more about what we believe makes a pull request complete, see our
[definition of done].

[documentation]: https://github.com/crossplane/crossplane/tree/master/docs
[examples]: https://github.com/crossplane/crossplane/tree/master/cluster/examples
[definition of done]: https://github.com/crossplane/crossplane/tree/master/design/one-pager-definition-of-done.md

### CODE REVIEW

File: docs/configure.md [hunk 1]
```diff
@@ -11,9 +11,15 @@ provider, you will need to create an account for Crossplane to use. Use the
...
+* [Google Cloud Platform (GCP) Service Account]
+* [Microsoft Azure Service Principal]
+* [Amazon Web Services (AWS) IAM User]
 
 Once you have configured your cloud provider account, you can get started
 provisioning resources!
+
+<!-- Named Links -->
+
+[Google Cloud Platform (GCP Service Account: cloud-providers/gcp/gcp-provider.md
```
[the remark this answers is not in the document]
Comment (hasheddan) [reply docs/configure.md]: Find and replace got me here, fixed :+1:

File: docs/app.md [hunk 2]
```diff
@@ -187,9 +180,9 @@ Now you are familiar with **Providers**, **Stacks**, and **Applications**. The
 next step is to build and deploy your own. Take a look at the following guides
 to learn more:
 
-- [Infrastructure Operators](1_infra_operators/1_installing_a_stack.md)
-- [Application Operators](2_app_operators/1_packaging_an_app.md)
-- [Developers](3_developers/1_requesting_infrastructure.md)
+- [Infrastructure Operators]
```
[the remark this answers is not in the document]
Comment (hasheddan) [reply docs/app.md:182]: These appear to have been broken already before this PR, I have updated with fixes :) thanks for catching!

**Comment by hasheddan** [comment]
@jbw976 see my first commit message on this PR haha. I **strongly** dislike named links. I have expressed this in a number of forums, but we are moving back to them here because of an issue with the jekyll plugin that generates our relative links in the docs (https://github.com/benbalter/jekyll-relative-links/issues/61#issue-589167883). It has been pretty consensus that folks want the markdown files to be line-wrapped for easier PR review, and unfortunately that frequently breaks the relative links. This PR is making the best of a bad situation unfortunately :disappointed:

File: docs/contributing/overview.md [hunk 3]
```diff
@@ -11,6 +11,12 @@ extend Crossplane. Please [open an
 issue](https://github.com/crossplane/crossplane/issues/new) for any additional
 documentation you would like to see in this section.
 
-1. [Services Developer Guide](1_services_developer_guide.md)
-2. [Observability Developer Guide](2_observability_developer_guide.md)
-3. [Troubleshoot](3_troubleshoot.md)
+1. [Services Developer Guide]
+2. [Observability Developer Guide]
+3. [Troubleshoot]
```
Comment (jbw976) [inline docs/contributing/overview.md:16]: this troubleshoot link is broken

File: docs/guides/guides.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+---
+
+# Guides
+
+Because of Crossplane's standardization on the Kubernetes API, it integrates
+well with many other project. Below is a collection of guides and tutorials that
+demonstrate how to use Crossplane with a variety of tools in the GitOps, service
+mesh, and infrastructure provisioning spaces.
+
+- [Connecting AWS managed services to your Argo CD pipeline with open source
```
Comment (jbw976) [inline docs/guides/guides.md:14]: this AWS/Argo link is broken

File: docs/contributing/overview.md (hunk 3 above)
Comment (hasheddan) [reply docs/contributing/overview.md]: Fixed :+1:

File: docs/guides/guides.md (hunk 4 above)
Comment (hasheddan) [reply docs/guides/guides.md:14]: Fixed :+1:

Comment (jbw976) [reply docs/guides/guides.md:14]: yayyy, now i can click this link and see this lovely article with the intro:

> This article is a guest post from Dan Mangum, a software engineer at Upbound.

🚀

**Pull request merged**